test/FastTests/Server/Replication/ReplicationResolveToLeader.cs

```diff
@@ -0,0 +1,291 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FastTests.Server.Basic.Entities;
+using Raven.NewClient.Client.Replication;
+using Xunit;
+
+namespace FastTests.Server.Replication
+{
+    public class ReplicationResolveToDatabase : ReplicationTestsBase
+    {
+        [Fact]
+        public async Task ResovleToDatabase()
+        {
+            var store1 = GetDocumentStore();
+            var store2 = GetDocumentStore();
+            using (var session = store1.OpenSession())
+            {
+                session.Store(new User{Name = "Karmel"},"foo/bar");
+                session.SaveChanges();
+            }
+            using (var session = store2.OpenSession())
+            {
+                session.Store(new User { Name = "Oren" }, "foo/bar");
+                session.SaveChanges();
+            }
+         
+            SetupReplication(store1,store2);
+
+            var conflicts = WaitUntilHasConflict(store2, "foo/bar");
+            Assert.Equal(2, conflicts["foo/bar"].Count);
+
+            SetupReplication(store2, new ReplicationDocument
+            {
+                DefaultResolver = new DatabaseResolver
+                {
+                    ResolvingDatabaseId = GetDocumentDatabaseInstanceFor(store2).Result.DbId.ToString(),
+                    Version = 0
+                }
+            }, store1);
+
+            await Task.Delay(500);
```

Comment:
I know, But have to wait until all changes are propagated (actually, this doesn't grantee it either).